server.js

```diff
@@ -602,10 +615,29 @@ function SocketIOFileUploadServer(options) {
 			socket.on(self.topicName + "_progress", _uploadProgress(socket));
 			socket.on(self.topicName + "_done", _uploadDone(socket));
 		}
+		self.functions.disconnect[socket.id] = _onDisconnect(socket);
+		socket.on("disconnect", self.functions.disconnect[socket.id]);
+	};
 
-		socket.on("disconnect", _onDisconnect(socket));
+	/**
+	 * Public method.  Undo listening from listen method.
+	 *
+	 * @param  {Socket} socket The socket on which not to listen anymore
+	 * @return {void}
+	 */
+	this.close = function (socket) {
+		if(_isWrapDataWellConfigured() && self.wrapData)
+			socket.removeAllListeners(self.topicName);
+		else {
+			socket.removeAllListeners(self.topicName + "_start");
+			socket.removeAllListeners(self.topicName + "_progress");
+			socket.removeAllListeners(self.topicName + "_done");
+		}
+		socket.off("disconnect", self.functions.disconnect[socket.id]);
+		self.functions.disconnect[socket.id]();
```

Comment:
One more little thing:

```javascript
delete self.functions.disconnect[socket.id];
```

Otherwise you will leak memory by holding a reference to an obsolete socket.

Reply:
yes, thanks for reviewing! I will add and push this in a few minutes.